src/p2/interpreter.rs

```diff
@@ -741,6 +737,157 @@ impl ParsedDocument {
+                dbg!(
+                    &subsection.name,
+                    &subsection.caption,
+                    &subsection.is_commented
+                );
+
+                for (_ln, key, value) in subsection.header.0.iter() {
+                    dbg!(key, value);
+                }
+
+                dbg!(&subsection.body);
+            }
+        }
+    }
+
+    /// same logic as [`ParsedDocument::ignore_comments()`] but uses the
+    /// already existing [`remove_comments()`] to remove comments.
+    ///
+    /// [`ParsedDocument::ignore_comments()`]: ftd::p2::interpreter::ParsedDocument::ignore_comments
+    /// [`Section::remove_comments()`]: ftd::p1::section::Section::remove_comments
+    #[allow(dead_code)]
+    fn ignore_all_comments(&mut self) -> ftd::p1::Result<()> {
+        self.sections = self
+            .sections
+            .iter()
+            .filter(|s| !s.is_commented)
+            .map(|s| s.remove_comments())
+            .collect::<Vec<ftd::p1::Section>>();
+        Ok(())
+    }
+
+    /// Filters out commented parts from the parsed document.
+    ///
+    /// # Comments are ignored for
+    /// 1.  /-- section: caption
+    ///
+    /// 2.  /section-header: value
+    ///
+    /// 3.  /body
+    ///
+    /// 4.  /--- subsection: caption
+    ///
+    /// 5.  /sub-section-header: value
+    ///
+    /// ## Note: To allow ["/content"] inside body, use ["\\/content"].
+    ///
+    /// Only '/' comments are ignored here.
+    /// ';' comments are ignored inside the [`parser`] itself.
+    ///
+    /// [`parser`]: ftd::p1::parser::parse
+    #[allow(dead_code)]
+    fn ignore_comments(&mut self) -> ftd::p1::Result<()> {
+        let mut processed_sections: Vec<ftd::p1::Section> = Vec::new();
+        for section in self.sections.iter_mut() {
+            // Ignore entire section if it is commented out
+            if section.is_commented {
+                continue;
+            }
+
+            // Ignore commented Headers
+            let mut header_list: ftd::p1::Header = Default::default();
+            for (ln, key, val) in section.header.0.iter() {
+                if key.starts_with('/') {
+                    continue;
+                }
+                header_list.0.push((*ln, key.to_string(), val.to_string()));
+            }
+
+            // Filtered header
+            section.header = header_list;
+
+            // Filter section body
+            section.body = {
+                match section.body {
+                    Some(ref b) if b.1.trim().is_empty() => None,
+                    Some(ref b) if b.1.trim().starts_with('/') => None,
+                    // To allow '/content' in section body, we need to use "\/content"
+                    // while stripping out the initial '\' from this body
+                    Some(ref b) if b.1.trim().starts_with(r"\/") => {
+                        Some((b.0, b.1.trim().replacen(r"\", "", 1)))
+                    }
```

Comment:
Check this for `headers` and `caption` too. Also use one function which does this in all headers, body and caption in sections and subsections.

Reply:
Done